docs/usage/network_policies.md

```diff
@@ -0,0 +1,148 @@
+# Network Policies in Gardener
```

Comment:
Can you please add the document to the index (docs/README.md)?

Reply:
Also, should this document be moved to `docs/extensions`? Or `docs/development`? It's not really related to the "usage of Gardener" (other than the small section about the network policies in the shoot because that's actually what might be interesting to end-users using shoots)?

Reply:
I was also unsure where to place it. I guess `docs/development` makes more sense